### ISSUE TITLE
Add `XLNetBackbone` to `models/`

### ISSUE THREAD

**Issue by shivance** [keras-team/keras-nlp#753]
XLNet was a new pretraining method for NLP, released in 2020, that significantly improves upon BERT on 20 tasks (e.g., SQuAD, GLUE, RACE). I think it deserves a place in Keras NLP's model zoo !

[arxiv](https://arxiv.org/abs/1906.08237)
[Github](https://github.com/zihangdai/xlnet) 
[Pretrained Weights](https://github.com/zihangdai/xlnet#pre-trained-models)

**Comment by shivance** [comment]
cc: @jbischof @chenmoneygithub @mattdangerw

**Comment by mattdangerw** [comment]
XLNet is probably something we would like eventually, but the token classification guide discussed on the other issue would be higher priority for us right now! We got lot's of backbones, but not enough examples :)

**Comment by susnato** [comment]
Hi @mattdangerw I would like to contribute this model, is it open to contribution?

**Comment by shivance** [comment]
Hi @susnato XLNet doesn't seem to be planned just yet. Once it's planned, @mattdangerw will add contribution welcome label, and I'll start working on it.

**Comment by shivance** [comment]
Opened PR for XLNet while #753 is pending for review.
